packages/lib/src/components/MBWay/components/MBWayInput/MBWayInput.tsx

```diff
@@ -1,69 +1,74 @@
+                const mappedCountries = filteredCountries.map(item => {
+                    // Get country flags (magic! - shifts the country code characters to the correct position of the emoji in the unicode space)
+                    const codePoints: number[] = item.id
+                        .toUpperCase()
+                        .split('')
+                        .map(char => 127397 + char.charCodeAt(0));
+
+                    // Get flag emoji + space at end (it doesn't work to add spaces in the template literal, below)
+                    const flag = String.fromCodePoint ? String.fromCodePoint(...codePoints) + '\u00A0\u00A0' : '';
+
+                    return { id: item.prefix, name: `${flag} ${item.prefix} (${item.id})`, selectedOptionName: `${flag} ${item.prefix}` };
+                });
+
+                setPhonePrefixes(mappedCountries || []);
+                setStatus('ready');
+            })
+            .catch(error => {
+                props.onError(new AdyenCheckoutError('ERROR', error));
+                setPhonePrefixes([]);
+                setStatus('ready');
+            });
+    }, []);
 
-    useEffect(() => {
+    const onChange = ({ data, valid, errors, isValid }) => {
         props.onChange({ data, valid, errors, isValid });
-    }, [data, valid, errors, isValid]);
+    };
 
     return (
-        <div className="adyen-checkout__mb-way">
-            <Field
-                errorMessage={!!errors.telephoneNumber && i18n.get('mobileNumber.invalid')}
-                label={i18n.get('mobileNumber')}
-                className={classNames('adyen-checkout__input--phone-number')}
-                isValid={valid.telephoneNumber}
-                dir={'ltr'}
-                name={'telephoneNumber'}
-            >
-                {renderFormField('tel', {
-                    value: data.telephoneNumber,
-                    className: 'adyen-checkout__pm__phoneNumber__input',
-                    placeholder: props.placeholders.telephoneNumber,
-                    required: true,
-                    autoCorrect: 'off',
-                    onBlur: handleChangeFor('telephoneNumber', 'blur'),
-                    onInput: handleChangeFor('telephoneNumber', 'input')
-                })}
-            </Field>
+        <LoadingWrapper status={status}>
+            <div className="adyen-checkout__mb-way">
+                <PhoneInput {...props} items={phonePrefixes} ref={phoneInputRef} onChange={onChange} data={props.data} />
 
-            {props.showPayButton && props.payButton({ status, label: i18n.get('confirmPurchase') })}
-        </div>
+                {props.showPayButton && props.payButton({ status, label: i18n.get('confirmPurchase') })}
+            </div>
+        </LoadingWrapper>
     );
 }
 
 MBWayInput.defaultProps = {
-    onChange: () => {}
+    onChange: () => {},
+    phoneNumberKey: 'mobileNumber',
+    phoneNumberErrorKey: 'mobileNumber.invalid'
```

Comment:
Do we support such customization for i18n? Is it needed?

Reply:
I think so - we might want to say "telephone number (invalid)" or "phone number (invalid)" or "mobile phone number (invalid)". 
It offers flexibility depending on what exactly the component is being used for and what the Ux-ers want to show for labels/error messages